src/components/Onfido/BaseOnfidoWeb.js

```diff
@@ -17,118 +17,135 @@ const propTypes = {
+            onSuccess(data);
+        },
+        onError: (error) => {
+            const errorMessage = lodashGet(error, 'message', CONST.ERROR.UNKNOWN_ERROR);
+            const errorType = lodashGet(error, 'type');
+            Log.hmmm('Onfido error', {errorType, errorMessage});
+            onError(errorMessage);
+        },
+        onUserExit: (userExitCode) => {
+            Log.hmmm('Onfido user exits the flow', {userExitCode});
+            onUserExit(userExitCode);
+        },
+        onModalRequestClose: () => {
+            Log.hmmm('Onfido user closed the modal');
+        },
+        language: {
+            // We need to use ES_ES as locale key because the key `ES` is not a valid config key for Onfido
+            locale: preferredLocale === CONST.LOCALES.ES ? CONST.LOCALES.ES_ES_ONFIDO : preferredLocale,
 
-                // Provide a custom phrase for the back button so that the first letter is capitalized,
-                // and translate the phrase while we're at it. See the issue and documentation for more context.
-                // https://github.com/Expensify/App/issues/17244
-                // https://documentation.onfido.com/sdk/web/#custom-languages
-                phrases: {
-                    'generic.back': this.props.translate('common.back'),
-                },
+            // Provide a custom phrase for the back button so that the first letter is capitalized,
+            // and translate the phrase while we're at it. See the issue and documentation for more context.
+            // https://github.com/Expensify/App/issues/17244
+            // https://documentation.onfido.com/sdk/web/#custom-languages
+            phrases: {
+                'generic.back': translate('common.back'),
             },
-        });
+        },
+    });
+}
 
-        window.addEventListener('userAnalyticsEvent', (event) => {
-            Log.hmmm('Receiving Onfido analytic event', event.detail);
+const Onfido = forwardRef((props, ref) => {
+    useEffect(() => {
+        initializeOnfido({
+            sdkToken: props.sdkToken,
+            onSuccess: props.onSuccess,
+            onError: props.onError,
+            onUserExit: props.onUserExit,
+            preferredLocale: props.preferredLocale,
+            translate: props.translate,
         });
-    }
 
-    render() {
-        return <div id={CONST.ONFIDO.CONTAINER_ID} />;
-    }
-}
+        const logOnFidoEvent = (event) => Log.hmmm('Receiving Onfido analytic event', event.detail);
+
+        window.addEventListener('userAnalyticsEvent', logOnFidoEvent);
+        return () => window.removeEventListener('userAnalyticsEvent', logOnFidoEvent);
+        // Onfido should be initialized only once on mount
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return (
+        <div
+            id={CONST.ONFIDO.CONTAINER_ID}
+            ref={ref}
+        />
+    );
+});
 
 Onfido.propTypes = propTypes;
 export default withLocalize(Onfido);
```

Comment:
Please use the `useLocalize` hook instead of `withLocalize` (and remove `withLocalize` props from the propTypes)